geth/jail/cell.go

```diff
@@ -48,3 +56,19 @@ func registerVMHandlers(v *vm.VM, lo *loop.Loop) error {
 
 	return nil
 }
+
+// Stop halts event loop associated with cell.
+func (c *Cell) Stop() {
+	c.cancel()
+}
+
+// CallAsync puts otto's function with given args into
+// event queue loop and schedules for immediate execution.
+// Intended to be used by any cell user that want's to run
+// async call, like callback.
+func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
```

Comment:
Can you describe what problems putting a callback into a loop solves?

Reply:
@adambabik do you mean describe here or in function comment?

Reply:
> It allows putting callback functions into safe execution in the event loop, rather then firing them in a separate goroutine.

Ok, sounds good. You removed some explicit locks thanks to this.